src/routes/addons.ts

```diff
@@ -6,28 +6,34 @@
  * (Department of Information and Computing Sciences)
  */
 
-import { Request, Response } from "express";
 import { Context } from "../context";
-import { AddonCategory } from "prisma/prisma-client";
+import { Addon, AddonCategory } from "@prisma/client";
 import { join } from "node:path";
+import { z } from "zod";
 
 // TODO: move this to a better place
 const pageSize = 20;
 
-/**
- * The request type for getting addons.
- */
+////////////////////////////////////////////////////////////////////////////////
 
-interface GetAddonsRequest extends Request {
-  query: {
-    page: number;
-    category?: AddonCategory;
-  };
+interface GetAddonsRequest {
+  page: number;
+  category?: AddonCategory;
+}
+
+const getAddonsSchema = z.object({
+  page: z.coerce.number().int().gte(0).default(0),
```

Comment:
You need `.coerce` because zod by default checks the type of the variable. The values in `req` are always strings, so validating `page` will always error without `.coerce`.